pkg/router/template/util/util.go

```diff
@@ -94,3 +104,12 @@ func GenerateBackendNamePrefix(termination routev1.TLSTerminationType) string {
 
 	return prefix
 }
+
+// HaproxyMaxTimeoutDuration is HaproxyMaxTimeout as a time.Duration value.
+var HaproxyMaxTimeoutDuration = func() time.Duration {
+	d, err := haproxytime.ParseDuration(HaproxyMaxTimeout)
+	if err != nil {
+		panic(err)
+	}
+	return d
+}()
```

Comment:
My concern here is that this a delayed failure. Why not parse the value in init and assign to a variable if all is OK.potentially

Reply:
If I understand https://go.dev/ref/spec#Package_initialization and https://go.dev/ref/spec#Program_initialization correctly, package variables (including `HaproxyMaxTimeoutDuration`) get initialized when the program starts.  Using `init` would be a slightly more verbose way to do effectively the same thing.  (In fact, using `init` would cause the initialization to happen slightly later, per the specification of package initialization that I referenced: "The entire package is initialized by assigning initial values to all its package-level variables followed by calling all `init` functions in the order they appear in the source, possibly in multiple files, as presented to the compiler.)

Reply:
ok. Thanks for the links.

Reply:
done